src/EventListener/DataContainer/LeadLabelListener.php

```diff
@@ -48,10 +48,13 @@ public function __invoke(array $row, string $label): string
         $records = $this->connection->fetchAllAssociative('SELECT name, value, label FROM tl_lead_data WHERE pid=?', [$row['id']]);
 
         foreach ($records as $record) {
+            $label = StringUtil::deserialize($record['label']);
+            $value = (!\is_array($label) && !empty($label)) || (\is_array($label) && array_filter($label)) ? $label : StringUtil::deserialize($record['value']);
```

Comment:
The empty check would ignore a value of `0`, which seems incorrect to me 🤔

Reply:
what would you suggest? my problem is that since the label fallback my lists are not working the way the did before this massive change!